MathLib/CMakeLists.txt

```diff
@@ -56,25 +56,29 @@ if(METIS_FOUND)
 endif()
 
 # Create the library
-add_library(MathLib STATIC ${SOURCES})
+add_library(MathLib ${SOURCES})
 
 set_target_properties(MathLib PROPERTIES LINKER_LANGUAGE CXX)
 
-target_link_libraries(MathLib INTERFACE
-	GeoLib
-	logog
+target_link_libraries(MathLib
+    #AssemblerLib   # Needed, but causes circular dependency.
+    logog
 )
 
+if(METIS_FOUND)
```

Comment:
TOdo for later, not now: we shouldnt always link metis and lapack just because Cmake could find them. there should be a flag like OGS_USE_METIS.